### PR TITLE
feat: Allow users to enter 12, 18 and 24 phrase mnemonics

### PR DESCRIPTION
**Pull request by alexandreaco**
Users can now select seed length options from a dropdown when creating and restoring a wallet s.

Screen Shot 2021-11-08 at 8 43 32 AM

Creating a wallet: https://www.loom.com/share/685ec18951cb4cf7b2d06ef2a6ee7780
Restoring a wallet: https://www.loom.com/share/e22d659f81bb4fc7b17464fa79af24de

### CODE REVIEW

**Review by saminakh — CHANGES_REQUESTED** [review]
👋 I tested wallet creation and restoration and those look good! I did note one spot that may have been missed--revealing the seed phrase while already authenticated:
Screen Shot 2021-11-09 at 3 09 23 PM
It looks like the seed phrase is correct but the count in the copy is still assuming 12.

I'll do a proper code review after this!

File: src/views/RestoreWallet/RestoreWalletEnterMnemonic.vue [hunk 1]
```diff
@@ -1,53 +1,117 @@
...
     MnemonicInput
   },
 
-  data () {
-    return {
-      inputWords: new Array(12).fill('') as string[]
+  setup () {
+    const { t } = useI18n({ useScope: 'global' })
+    const mnemonicStrength: Ref<StrengthT> = ref(StrengthT.WORD_COUNT_12)
+    const strengthOptions: AppSelectOptionT[] = [
```
Comment (saminakh) [inline src/views/RestoreWallet/RestoreWalletEnterMnemonic.vue:52]: This part (and some other code in this page) feels kind of repetitive with code in `CreateWalletViewMnemonic.vue`, and potentially some of the code in the 'reveal seed phrase' view too. Maybe it'd be worth breaking some of this into a mnemonic helper? Some of the logic to build the select or something? Just a thought though! You're the vue expert!

**Review by saminakh — COMMENTED** [review]
Left some Q's and comments!

**Review by staylorwr — CHANGES_REQUESTED** [review]
https://www.loom.com/share/c441ba9c9eca405f9c4f1041515e4702

@alexandreaco functionally this looks great.  

In the loom above, could you update the padding or layout of the page so that:
- toggling show / hide state of the 'change seed phase length' ui doesn't change heights?  Maybe just set a fixed height of the parent element so it doesn't change the page layout?
- update the vertical padding of the 24-word phrase so it doesn't push the button below the fold.

**Comment by alexandreaco** [comment]
Thanks for the reviews!

@staylorwr Updated padding on create https://www.loom.com/share/0ace9507761f438081fb044633035cee
@saminak Added the UI missing work for viewing a user's current mnemonic
Screen Shot 2021-11-29 at 9 59 04 AM

**Review by staylorwr — APPROVED** [review]
🎁

**Review by saminakh — CHANGES_REQUESTED** [review]
It looks like the `mnemonicDisclaimer` in `src/text/index.ts` still has the '12' hardcoded in it, so on the `SettingsRevealMnemonic.vue` page it's not always displaying the right count in that copy.

File: src/components/AppSelect.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    >
+      <span class="block truncate flex-1">
+        {{ selected.label }}
+      </span>
+      <svg width="14" height="8" viewBox="0 0 14 8" fill="none" xmlns="http://www.w3.org/2000/svg">
+        <path d="M1 1L7 7L13 1" class="stroke-current text-rBlack" stroke-linecap="round" stroke-linejoin="round"/>
+      </svg>
+    </button>
+
+    <ul
```
[the remark this answers is not in the document]
Comment (alexandreaco) [reply src/components/AppSelect.vue:15]: Unfortunately, that's not a feature that comes out-of-the-box with vue. Alpine is pretty incredible for including that as a default. I'll bring this up to the radix team when then QA it so see if we should invest a bit more time to set this up

**Comment by alexandreaco** [comment]
@saminakh Fixed, thank you for your eagle eyes 🙇‍♀️

File: src/helpers/mnemonic.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  },
+  {
+    id: StrengthT.WORD_COUNT_18,
+    label: '18 word seed phrase'
+  },
+  {
+    id: StrengthT.WORD_COUNT_24,
+    label: '24 word seed phrase'
+  }
+]
```
Comment (saminakh) [inline src/helpers/mnemonic.ts:17]:
```
 ERROR  Failed to compile with 1 error                                                                                                                                                                           11:45:08 AM

 error  in ./src/helpers/mnemonic.ts

Module Error (from ./node_modules/eslint-loader/index.js):

/Users/samina/Documents/radix/olympia-wallet/src/helpers/mnemonic.ts
  17:2  error  Newline required at end of file but not found  eol-last

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```
  😭 This linter has no chill

**Review by saminakh — CHANGES_REQUESTED** [review]
one last linter shenanigan, sorry alex :(

**Comment by alexandreaco** [comment]
That newline error 🤦‍♀️ I see the updated copy when I run it, can you confirm one more time?

Screen Shot 2021-12-02 at 10 59 38 AM

**Review by saminakh — APPROVED** [review]
🎉 LGTM! Thank you for all the fixes!

**Pull request merged**